Mathlib/Topology/Algebra/Module/ClosedSubmodule.lean

```diff
@@ -0,0 +1,179 @@
+    (s.comap g).comap f = s.comap (g.comp f) := rfl
+
+instance instInf : Min (ClosedSubmodule R M) where
+  min s t := ⟨s ⊓ t, s.isClosed.inter t.isClosed⟩
+
+instance instInfSet : InfSet (ClosedSubmodule R M) where
+  sInf S := ⟨⨅ s ∈ S, s, by simpa using isClosed_biInter fun x hx ↦ x.isClosed⟩
+
+@[simp, norm_cast]
+lemma toSubmodule_sInf (S : Set (ClosedSubmodule R M)) :
+    toSubmodule (sInf S) = ⨅ s ∈ S, s.toSubmodule := rfl
+
+@[simp, norm_cast]
+lemma toSubmodule_iInf (f : ι → ClosedSubmodule R M) :
+    toSubmodule (⨅ i, f i) = ⨅ i, (f i).toSubmodule := by rw [iInf, toSubmodule_sInf, iInf_range]
+
+@[simp, norm_cast]
+lemma coe_sInf (S : Set (ClosedSubmodule R M)) : ↑(sInf S) = ⨅ s ∈ S, (s : Set M) := by
+  simp [← coe_toSubmodule]
+
+@[simp, norm_cast]
+lemma coe_iInf (f : ι → ClosedSubmodule R M) : ↑(⨅ i, f i) = ⨅ i, (f i : Set M) := by
+  simp [← coe_toSubmodule]
+
+@[simp] lemma mem_sInf {S : Set (ClosedSubmodule R M)} : x ∈ sInf S ↔ ∀ s ∈ S, x ∈ s := by
+  simp [← SetLike.mem_coe]
+
+@[simp] lemma mem_iInf {f : ι → ClosedSubmodule R M} : x ∈ ⨅ i, f i ↔ ∀ i, x ∈ f i := by
+  simp [← SetLike.mem_coe]
+
+instance instSemilatticeInf : SemilatticeInf (ClosedSubmodule R M) :=
+  toSubmodule_injective.semilatticeInf _ fun _ _ ↦ rfl
+
+@[simp, norm_cast]
+lemma toSubmodule_inf (s t : ClosedSubmodule R M) :
+    toSubmodule (s ⊓ t) = s.toSubmodule ⊓ t.toSubmodule := rfl
+
+@[simp, norm_cast] lemma coe_inf (s t : ClosedSubmodule R M) : ↑(s ⊓ t) = (s ⊓ t : Set M) := rfl
+
+@[simp] lemma mem_inf : x ∈ s ⊓ t ↔ x ∈ s ∧ x ∈ t := .rfl
+
+instance instTop : Top (ClosedSubmodule R M) where top := ⟨⊤, isClosed_univ⟩
+
+@[simp, norm_cast] lemma toSubmodule_top : toSubmodule (⊤ : ClosedSubmodule R M) = ⊤ := rfl
+
+@[simp, norm_cast] lemma coe_top : ((⊤ : ClosedSubmodule R M) : Set M) = .univ := rfl
+
+@[simp] lemma mem_top : x ∈ (⊤ : ClosedSubmodule R M) := trivial
+
+section T1Space
+variable [T1Space M]
+
+instance instOrderBot : OrderBot (ClosedSubmodule R M) where
+  bot := ⟨⊥, isClosed_singleton⟩
+  bot_le s := bot_le (a := s.toSubmodule)
+
+@[simp, norm_cast] lemma toSubmodule_bot : toSubmodule (⊥ : ClosedSubmodule R M) = ⊥ := rfl
+
+@[simp, norm_cast] lemma coe_bot : ((⊥ : ClosedSubmodule R M) : Set M) = {0} := rfl
+
+@[simp] lemma mem_bot : x ∈ (⊥ : ClosedSubmodule R M) ↔ x = 0 := .rfl
+
+end T1Space
+end ClosedSubmodule
+
+namespace Submodule
+variable [ContinuousAdd M] [ContinuousConstSMul R M]
+
+/-- The closure of a submodule as a closed submodule. -/
+@[simps!]
+protected def closure (s : Submodule R M) : ClosedSubmodule R M where
+  toSubmodule := s.topologicalClosure
+  isClosed' := isClosed_closure
+
+end Submodule
+
+namespace ClosedSubmodule
+variable [ContinuousAdd N] [ContinuousConstSMul R N]
+
+/-- The closure of image of a proper cone under a continuous `R`-linear map is a proper cone. We
+use continuous maps here so that the comap of f is also a map between proper cones. -/
```

Comment:
```suggestion
/-- The closure of image of a closed submodule under a continuous linear map is a closed submodule.
We use continuous maps here so that the `comap` of f is also a map between closed submodules. -/
```